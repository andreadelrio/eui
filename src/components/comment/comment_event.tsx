import React, { FunctionComponent, ReactNode } from 'react';
import { CommonProps, keysOf } from '../common';
import classNames from 'classnames';

export type EuiCommentEventProps = CommonProps &
  CommonProps & {
    /**
     * Author of the comment. Display a small icon or avatar with it if needed.
     */
    username: ReactNode;
    timestamp?: ReactNode;
    /**
     * Describes the event that took place
     */
    event?: ReactNode;
    /**
     * Actions the user can perform from the comment's header
     */
    actions?: ReactNode;
    /**
     * Use "update" when the comment is primarily showing info about actions that the user or the system has performed (e.g. "user1 edited a case").
     */
    type?: EuiCommentType;
  };

const typeToClassNameMap = {
  regular: 'euiCommentEvent--regular',
  update: 'euiCommentEvent--update',
};

export const TYPES = keysOf(typeToClassNameMap);
export type EuiCommentType = keyof typeof typeToClassNameMap;

export const EuiCommentEvent: FunctionComponent<EuiCommentEventProps> = ({
  children,
  className,
  username,
  timestamp,
  type = 'regular',
  event,
  actions,
  ...rest
}) => {
  const classes = classNames(
    'euiCommentEvent',
    typeToClassNameMap[type],
    className
  );

  return (
    <div data-focusable="true" className={classes} {...rest}>
      <div className="euiCommentEvent__header">
        <div className="euiCommentEvent__headerData">
          <div className="euiCommentEvent__headerUsername">{username}</div>
          <div className="euiCommentEvent__headerEvent">{event}</div>
          <div className="euiCommentEvent__headerTimestamp">{timestamp}</div>
        </div>
        <div className="euiCommentEvent__headerActions">{actions}</div>
      </div>
      <div className="euiCommentEvent__body">{children}</div>
    </div>
  );
};
