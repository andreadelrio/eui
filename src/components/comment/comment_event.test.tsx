import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test/required_props';

import { EuiCommentEvent } from './comment_event';

describe('EuiCommentEvent', () => {
  test('is rendered', () => {
    const component = render(
      <EuiCommentEvent username="someuser" {...requiredProps} />
    );

    expect(component).toMatchSnapshot();
  });
});
