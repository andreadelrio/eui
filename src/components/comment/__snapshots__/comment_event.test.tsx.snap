// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiCommentEvent is rendered 1`] = `
<div
  aria-label="aria-label"
  class="euiCommentEvent euiCommentEvent--regular testClass1 testClass2"
  data-test-subj="test subject string"
>
  <div
    class="euiCommentEvent__header"
  >
    <div
      class="euiCommentEvent__headerData"
    >
      <div
        class="euiCommentEvent__headerUsername"
      >
        someuser
      </div>
      <div
        class="euiCommentEvent__headerEvent"
      />
    </div>
  </div>
</div>
`;
